check-cla/action.yml

```diff
@@ -3,161 +3,196 @@ name: CLA check
       uses: actions/github-script@v6
-      id: contributors
+      id: metadata
       with:
         github-token: ${{ inputs.token }}
         script: |
-          console.log(context);
-          const getContributors = async () => {
-            try {
-              const results = (
-                await github.rest.repos.getContent({
-                  owner: 'conda',
-                  repo: 'infrastructure',
-                  path: '.clabot'
-                })
-              );
-              return JSON.parse(Buffer.from(results.data.content, results.data.encoding).toString('utf-8')).contributors;
-            } catch (err) {
-              core.error(`Could not retrieve contributors, returning undefined. Reason: ${err}`)
-              return undefined;
-            }
-          }
-          const contributors = await getContributors();
-          console.log(contributors);
-          const pull_request = (context.payload.issue || context.payload.pull_request || context.payload);
-          const creator = pull_request.user.login;
-          console.log(creator);
-          const hasSigned = contributors.includes(creator);
-          console.log(hasSigned);
-          core.setOutput('contributors', contributors);
-          core.setOutput('hasSigned', hasSigned);
-
-    # add [cla-signed] label if actor has already signed
-    - name: Add label
+          const { owner, repo, number } = context.issue;
+          core.debug(`owner: ${owner}`);
+          core.debug(`repo: ${repo}`);
+          core.setOutput('number', number);
+          core.debug(`number: ${number}`);
+
+          const raw = await github.rest.pulls.get({
+            owner: owner,
+            repo: repo,
+            pull_number: number
+          });
+          const labels = raw.data.labels.map(label => label.name);
+          core.debug(`labels: ${labels}`);
+
+          const has_label = labels.includes('${{ inputs.label }}');
+          core.setOutput('has_label', has_label);
+          core.debug(`has_label: ${has_label}`);
+
+          const { content, encoding } = (await github.rest.repos.getContent({
+            owner: owner,
+            repo: repo,
+            path: '${{ inputs.cla_path }}'
+          })).data;
+          const contributors = JSON.parse(
+            Buffer.from(content, encoding).toString('utf-8')
+          ).contributors;
+          core.debug(`contributors: ${contributors}`);
+
+          const payload = context.payload.issue || context.payload.pull_request || context.payload;
+          const contributor = payload.user.login;
+          core.setOutput('contributor', contributor);
+          core.debug(`contributor: ${contributor}`);
+
+          const url = payload.html_url;
+          core.setOutput('url', url);
+          core.debug(`url: ${url}`);
+
+          const has_signed = contributors.includes(contributor);
+          core.setOutput('has_signed', has_signed);
+          core.debug(`has_signed: ${has_signed}`);
```

Comment:
Moved `sha` detection into new `set-commit-status` action. Removed `labels` output since we don't use it outside of this scope. Switch `hasLabel` to `has_label` for variable consistency.

Flattened logic since the error catching doesn't actually do anything special. Removes `contributors` output since we don't use it outside of this scope. Switch `hasSigned` to `has_signed` for variable naming consistency.

Reply:
Good changes!